docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,125 @@
+# thin-edge Domain Model
+
+The following section introduces the **thin-edge** domain model.
+
+The **thin-edge** domain model explains details of target domains **thin-edge** is primarily designed for.
+It identifies _entities_ and _aspects_ of those target domains, that are touch points for **thin-edge**.
+Additionally it introduces new _entities_ or _aspects_ to seamlessly fit **thin-edge** into those target domains.
+
+Finally the **thin-edge** domain model gives..
+  * _target domain experts_ a clear idea how to position **thin-edge** in their target domain
+  * _target domain experts_ **and** _thin-edge developers_ a common understanding and unique vocabulary for **thin-edge** and it's environment
+
+## thin-edge target domains
+
+**thin-edge** is designed to facilitate IoT functionality to resource constrained **devices**.
+The focus is on industrial OT **devices** or any other kind of embedded **devices**. It is not reduced to **devices** that are capable to install and run thin-edge, but includes also **devices** that need another _(gateway) device_ aside, that executes **thin-edge**.
+
+Usual **devices** are **PLCs** (**P**rogrammable **L**ogic **C**ontrollers), **IPCs** (**I**ndustrial **PC**s) or any kind of **SoC-based** or **Microcontroller-based** Embedded System. The figure below shows a simplified conceptual model of such a device.
+
+![Simple Device Model](images/simple-device-model.svg)
+
+<!--
+* TODO: add somehow "Such a **device** is most often a specialized hardware that has been ruggedized to operate in harsh electrical, thermic and mechanical industrial environments."
+-->
+* The **Domain Application** is a program, usually developed by a **Domain Expert**
```

Comment:
Lets use the same user persona's as defined in the vision document

Reply:
```suggestion
* The **Domain Application** is a program, usually developed by an **End User** (e.g. a PLC Application Engineer or an Embedded Software Developer)
```


@itsroshan , do you agree to that?

Reply:
The end-user is not supposed to develop something ;-) According to the [current vision document](https://github.com/thin-edge/thin-edge.io/pull/1226/files) this is done by an "Interface Developer". That said using this term here seems a bit weird. This is for me a sign that the vision should be updated with a better name.

Reply:
For me it is even not the "Interface Developer". That is since the one who implements by example a **PLC application** (the domain application on a PLC) might get never in touch with thin-edge. Instead some **protocol driver** might gather data that the **PLC application** just provides.
In that case the "Interface Developer" is the one who implements the **protocol driver**. However I want to highlight here the guy who has all the knowledge about the domain specific details (in case of PLCs it is the **PLC automation engineer** who has expert knowledge about the automation process).

I feel we miss that persona that I refer here as **Domain Expert** in the current vision document.

Reply:
> If I understand correctly your comment, the work of this domain expert is unrelated to thin-edge. Am I correct? If so, there is no point to say a work about him in the vision or even this doc.

That is true. I just want to explain who is in charge to create that application to give an idea to the reader what that application is about. However next lines refere to 'automation engineer' and 'embedded software engineer', so that fits that purpose.

So I will remove the persona here.